esphome/components/image/__init__.py

```diff
@@ -139,6 +192,19 @@ def validate_file_shorthand(value):
                 CONF_ICON: icon,
             }
         )
+    if value.startswith("https"):
```

Comment:
I'm also in favor of the "https only" Internet, but there are people still using non-encrypted http ;-) (Especially in local-only setups)
```suggestion
    if value.startswith("http://") or value.startswith("https://"):
```